subscriber.go

```diff
@@ -676,24 +676,13 @@ func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wr
 	}()
 
 	if wrapSel {
-		// Note this branch is nested under wrapSel because wrapSel adds the
-		// semantics that we stop when we hit the `h.latestSync` node. This is
-		// a special case where we are starting at the stop node so we can just
-		// return.
-		if h.latestSync != nil && h.latestSync.(cidlink.Link).Cid == nextCid {
-			// Nothing to do. We've already synced to this cid because we have
-			// it as h.latestSync.
-			log.Infow("Already synced")
-			return nil
-		}
-
 		sel = ExploreRecursiveWithStopNode(h.subscriber.syncRecLimit, sel, h.latestSync)
-	} else if checkAlreadySynced {
-		// Special case when starting at stop node; selector does not check it.
-		if h.latestSync != nil && h.latestSync.(cidlink.Link).Cid == nextCid {
-			// Nothing to do. We've already synced to this cid because we have
-			// it as h.latestSync.
-			log.Infow("Already synced")
+	}
+
+	if sel != nil {
```

Comment:
How can `sel` ever be `nil` here?.  If `wrapSel` is `true`, then `sel` gets assigned.  If `wrapSel` is false, then `sel` was already non-nil.